src/Symfony/Component/Config/Definition/EnumNode.php

```diff
@@ -78,13 +104,48 @@ protected function finalizeValue(mixed $value): mixed
     {
         $value = parent::finalizeValue($value);
 
-        if (!\in_array($value, $this->values, true)) {
-            $ex = new InvalidConfigurationException(\sprintf('The value %s is not allowed for path "%s". Permissible values: %s', json_encode($value), $this->getPath(), $this->getPermissibleValues(', ')));
-            $ex->setPath($this->getPath());
+        if ($this->enumFqcn) {
+            if (is_a($this->enumFqcn, \BackedEnum::class, true)) {
+                if (\is_string($value) || \is_int($value)) {
+                    try {
+                        $case = $this->enumFqcn::tryFrom($value);
+                    } catch (\TypeError) {
+                        throw new InvalidConfigurationException(\sprintf('The value could not be casted to a case of the "%s" enum. Is the value the same type as the backing type of the enum?', $this->enumFqcn));
```

Comment:
"casted" is incorrect in English
Don't we wan't to show the erroneous value in the message BTW?
```suggestion
                        throw new InvalidConfigurationException(\sprintf('The value could not be cast to a case of the "%s" enum. Is the value the same type as the backing type of the enum?', $this->enumFqcn));
```

Reply:
cast**ed*